src/app/util/af.h

```diff
@@ -569,13 +569,27 @@ uint8_t emberAfStringLength(const uint8_t * buffer);
  */
 uint16_t emberAfLongStringLength(const uint8_t * buffer);
 
+/*
+ * @brief Function that copies a ZCL typed list into a buffer. The index parameter
+ * may indicate a specific member of the list of the whole list if it is equal to -1.
```

Comment:
This documentation still does not make it clear that index == 0 is special and does weird things, and that in particular you have to provide it a different amount of memory than for other indices.  It's a footgun, and I'd really rather we did not have it.

Reply:
I have added some documentation. For reference I have mostly done this looking at the `ARRAY` definition of the ZCL spec.